File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=''.join(ch.lower() for ch in s if ch.isalnum())
        if len(s) == 0:
            return True
        i = 0
        j = len(s)-1
        res = True
        while(i<=j):
            if s[i]!= s[j]:
                return False
            else :
                i+=1
                j-=1
        return res

        

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String opr = s.replaceAll("[^a-zA-Z0-9]","").toLowerCase();
        int left = 0 ,right = opr.length()-1;
        while(left<right){
            if(opr.length()==1) {return true;}
             if (opr.charAt(left) != opr.charAt(right)){return false;}
            else{left++; right--;} }
        return true;
    }
}